src/basic/log.c

```diff
@@ -49,6 +49,12 @@ static void *log_syntax_callback_userdata = NULL;
 
 static LogTarget log_target = LOG_TARGET_CONSOLE;
 static int log_max_level = LOG_INFO;
+static int log_target_max_level[] = {
+        [LOG_TARGET_CONSOLE] = INT_MAX,
+        [LOG_TARGET_KMSG]    = INT_MAX,
+        [LOG_TARGET_SYSLOG]  = INT_MAX,
+        [LOG_TARGET_JOURNAL] = INT_MAX,
```

Comment:
Maybe, better to reorder the definition of LOG_TARGET_XYZ, and define these four earlier, and add _LOG_TARGET_SINGLE_FUNC or so to distinguish them from others like _JOURNAL_OR_KMSG ?